### PR TITLE
Optim-wip: Fix InceptionV1 model layers for attributions

### PR DESCRIPTION
**Pull request by ProGamerGov**
This PR should resolve the issue where the existing InceptionV1 model class does not give the correct attribution values when using Captum's attr module.

Changes made to the InceptionV1 model class:

* Removed `F.pad` in favour of `Conv2dSame` for the applicable `nn.Conv2d` layers.
* Added `ceil_mode=True` to the pooling layers.
* Changed the LocalResponseNorm layer values to match Ludwig / Clarity's values, to get the correct attributions. While testing I didn't notice any difference between the two visually, but I'm inclined to trust Ludwig's calculated values over the ones I got from Microsoft's MMdnn.

### CODE REVIEW

**Comment by ProGamerGov** [comment]
@NarineK This PR is just a small fix for the InceptionV1 model class and thus shouldn't interfere or overlap with https://github.com/pytorch/captum/pull/552 or any other PR, and thus can be merged at any time.

**Comment by NarineK** [comment]
@ProGamerGov, thank you for the fix! Are you using layer attribution ? How were you using it that it didn't give you correct answers ?

**Comment by ProGamerGov** [comment]
@NarineK Yes I'm using layer attribution for the channel attribution! The ReLU position thing was actually not the cause and thus I reverted that change. I think that the issue was mostly caused by the Local Response Norm values?


This is one of the tests I was using to determine the correct output (by comparing the results with Lucid):

```
import torch
import numpy as np
from PIL import Image
from captum import attr
from captum.optim._models.inception_v1 import googlenet

!wget https://storage.googleapis.com/lucid-static/building-blocks/examples/dog_cat.png
image = Image.open("dog_cat.png").convert('RGB')
catdog_image = torch.as_tensor(np.array(image)).permute(2, 0, 1).unsqueeze(0) / 255

model = googlenet(pretrained=True, replace_relus_with_redirectedrelu=False).eval()

# Channel Attribution
get_attr = attr.LayerGradientXActivation(attr_model, attr_model.mixed4d)
# Labrador retriever: 176, tiger cat: 55
attr_tensor = get_attr.attribute(catdog_image, 176) - get_attr.attribute(catdog_image, 55)
# Reduce down to channels
attr_tensor = attr_tensor[0].sum(1).sum(1)

# Values should be: 426, 436,  43, 477,  37, 139, 149, 425, 100,  77, 399, 418, 420, 101, ...
print(torch.argsort(-attr_tensor))
```

To determine what the correct values were I was printing the attribution outputs from the [Lucid Channel Attribution notebook](https://colab.research.google.com/github/tensorflow/lucid/blob/master/notebooks/building-blocks/AttrChannel.ipynb).

**Comment by NarineK** [comment]
> 176



> @NarineK Yes I'm using layer attribution for the channel attribution! The ReLU position thing was actually not the cause and thus I reverted that change. I think that the issue was mostly caused by the Local Response Norm values?
> 
> This is one of the tests I was using to determine the correct output (by comparing the results with Lucid):
> 
> ```
> import torch
> import numpy as np
> from PIL import Image
> from captum import attr
> from captum.optim._models.inception_v1 import googlenet
> 
> !wget https://storage.googleapis.com/lucid-static/building-blocks/examples/dog_cat.png
> image = Image.open("dog_cat.png").convert('RGB')
> catdog_image = torch.as_tensor(np.array(image)).permute(2, 0, 1).unsqueeze(0) / 255
> 
> model = googlenet(pretrained=True, replace_relus_with_redirectedrelu=False).eval()
> 
> # Channel Attribution
> get_attr = attr.LayerGradientXActivation(model, model.mixed4d)
> # Labrador retriever: 176, tiger cat: 55
> attr_tensor = get_attr.attribute(catdog_image, 176) - get_attr.attribute(catdog_image, 55)
> # Reduce down to channels
> attr_tensor = attr_tensor[0].sum(1).sum(1)
> 
> # Values should be: 426, 436,  43, 477,  37, 139, 149, 425, 100,  77, 399, 418, 420, 101, ...
> print(torch.argsort(-attr_tensor))
> ```
> 
> To determine what the correct values were I was printing the attribution outputs from the [Lucid Channel Attribution notebook](https://colab.research.google.com/github/tensorflow/lucid/blob/master/notebooks/building-blocks/AttrChannel.ipynb).

I see - thank you for the reference. In case of the gradient x activations, if model parameterizations match then gradients should be nearly identical for tensorflow and pytorch. There might be slight differences in terms of numerical instability but  they shouldn't be significant.
Another way to compare is to compare activation vectors and weight matrices in the layers; If they match then, grads should match too.

**Review by NarineK — APPROVED** [review]
LGTM! Thank you :)

**Comment by ProGamerGov** [comment]
@NarineK Awesome! The InceptionV1 model outputs for attributions should now be exactly the same as the TensorFlow version after this PR gets merged.
